packages/router/src/router.tsx

```diff
@@ -108,37 +95,35 @@ const InternalRoute: React.VFC<InternalRouteProps> = ({
     return <Redirect to={newPath} />
   }
 
-  if (!page || !name) {
-    throw new Error(
-      "A route that's not a redirect or notfound route needs to specify " +
-        'both a `page` and a `name`'
-    )
-  }
-
   const Page = activePageContext.loadingState[path]?.page || (() => null)
 
   // Level 3/3 (InternalRoute)
   return <Page {...allParams} />
 }
 
+/**
+ * A user-defined type guard.
+ *
+ * @see {@link https://www.typescriptlang.org/docs/handbook/2/narrowing.html#using-type-predicates}
+ * */
```

Comment:
When I see an explicitly typed return, my instinct is to remove it so added this here to ward off that knee-jerk reaction.